File: app/src/main/java/me/kirimin/mitsumine/db/NGWordDAO.java
package me.kirimin.mitsumine.db;

import java.util.ArrayList;
import java.util.List;

import me.kirimin.mitsumine.model.NGWord;

import com.activeandroid.query.Delete;
import com.activeandroid.query.Select;

public class NGWordDAO {

    public static void save(String word) {
        NGWord ngWord = new NGWord();
        ngWord.word = word;
        ngWord.save();
    }

    public static List<String> findAll() {
        List<NGWord> list = new Select().from(NGWord.class).execute();
        List<String> stringList = new ArrayList<>();
        for (NGWord w : list) {
            stringList.add(w.word);
        }
        return stringList;
    }

    public static void delete(String word) {
        new Delete().from(NGWord.class).where("word = ?", word).execute();
    }

    private NGWordDAO() {
    }
}

File: app/src/main/java/me/kirimin/mitsumine/db/NGWordDAO.kt
package me.kirimin.mitsumine.db

import com.activeandroid.Model
import java.util.ArrayList

import com.activeandroid.query.Delete
import com.activeandroid.query.Select
import me.kirimin.mitsumine.model.NGWord

public class NGWordDAO private constructor() {
    companion object {

        public fun save(word: String) {
            val ngWord = NGWord()
            ngWord.word = word
            ngWord.save()
        }

        public fun findAll(): List<String> {
            val list = Select().from(javaClass<NGWord>()).execute<NGWord>()
            val stringList = ArrayList<String>()
            for (w in list) {
                stringList.add(w.word)
            }
            return stringList
        }

        public fun delete(word: String) {
            Delete().from(javaClass<NGWord>()).where("word = ?", word).execute<Model>()
        }
    }
}
